xds/internal/server/listener_wrapper_test.go

```diff
@@ -21,154 +21,226 @@ package server
+//     address to which our net.Listener is bound to. Test verifies that the
+//     listenerWrapper does not become ready.
+//   - the management server returns a Listener resource that contains an inline
+//     Route Configuration. Test verifies that the listenerWrapper becomes ready.
+func (s) TestListenerWrapper_InlineRouteConfig(t *testing.T) {
+	mgmtServer, nodeID, ldsResourceNamesCh, _, xdsC := xdsSetupFoTests(t)
+	readyCh, host, port, lisResourceName := createListenerWrapper(t, xdsC)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	waitForResourceNames(ctx, t, ldsResourceNamesCh, []string{lisResourceName})
+
+	// Configure the management server with a listener resource that is expected
+	// to be NACKed.
+	resources := e2e.UpdateOptions{
+		NodeID:         nodeID,
+		Listeners:      []*v3listenerpb.Listener{badListenerResource(t, lisResourceName)},
+		SkipValidation: true,
+	}
+	if err := mgmtServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that there is no message on the ready channel.
+	sCtx, sCtxCancel := context.WithTimeout(ctx, defaultTestShortTimeout)
+	defer sCtxCancel()
+	select {
+	case <-readyCh:
+		t.Fatalf("Ready channel written to after receipt of a bad Listener update")
+	case <-sCtx.Done():
+	}
+
+	// Configure the management server with a listener resource that does not
+	// match the address to which our listener is bound to.
+	resources.Listeners = []*v3listenerpb.Listener{e2e.DefaultServerListener(host, port+1, e2e.SecurityLevelNone, route1)}
+	if err := mgmtServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that there is no message on the ready channel.
+	select {
+	case <-readyCh:
+		t.Fatalf("Ready channel written to after receipt of a bad Listener update")
+	case <-sCtx.Done():
+	}
+
+	// Configure the management server with a good update that does not contain
+	// any rds names that need reolution.
+	resources.Listeners = []*v3listenerpb.Listener{e2e.DefaultServerListener(host, port, e2e.SecurityLevelNone, route1)}
+	if err := mgmtServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the listener wrapper becomes ready.
+	select {
+	case <-ctx.Done():
+		t.Fatalf("Timeout waiting for the ready channel to be written to after receipt of a good Listener update")
+	case <-readyCh:
+	}
 }
 
-func Test(t *testing.T) {
-	grpctest.RunSubTests(t, s{})
+// Tests the case where a listenerWrapper is created and the management server
+// returns a Listener resource that specifies the name of a Route Configuration
+// resource. The test verifies that the listenerWrapper does not become ready
+// when waiting for the Route Configuration resource and becomes ready once it
+// receives the Route Configuration resource.
+func (s) TestListenerWrapper_RouteNames(t *testing.T) {
+	oldRBAC := envconfig.XDSRBAC
+	envconfig.XDSRBAC = true
+	defer func() {
+		envconfig.XDSRBAC = oldRBAC
+	}()
```

Comment:
Why does this need to turn on this env var? Is it so this conditional doesn't hit: https://github.com/grpc/grpc-go/blob/59f57b160e79284dee90b26e4532a57542eaf056/xds/internal/server/listener_wrapper.go#L267? If so, can you briefly mention why this is turned on. The test above does not turn it on.